java/client/src/main/java/glide/api/models/commands/bitmap/BitFieldOptions.java

```diff
@@ -0,0 +1,187 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.bitmap;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import glide.api.commands.BitmapBaseCommands;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Subcommand arguments for {@link BitmapBaseCommands#bitfield(String, BitFieldSubCommands[])} and
+ * {@link BitmapBaseCommands#bitfieldReadOnly(String, BitFieldReadOnlySubCommands[])}. Specifies
+ * subcommands, bit-encoding type, and offset type.
+ *
+ * @see <a href="https://redis.io/commands/bitfield/">redis.io</a>
+ */
+public class BitFieldOptions {
+    /** Subcommands for {@link BitmapBaseCommands#bitfield(String, BitFieldSubCommands[])}. */
+    public interface BitFieldSubCommands {
+        String[] createArgs();
```

Comment:
Rename to align with other options and add a doc.
```suggestion
        String[] toArgs();
```